src/sql/workbench/contrib/query/browser/queryResultsView.ts

```diff
@@ -291,41 +294,42 @@ export class QueryResultsView extends Disposable {
 		}
 	}
 
-	public set input(input: QueryResultsInput) {
+	public set input(input: QueryResultsInput | undefined) {
 		this._input = input;
-		this.runnerDisposables.dispose();
-		this.runnerDisposables = new DisposableStore();
+		this.runnerDisposables.clear();
```

Comment:
Assuming this is a helper which is replacing these two lines

Reply:
Yup

https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/common/lifecycle.ts#L139

```
/**
	 * Dispose of all registered disposables but do not mark this object as disposed.
	 */
	public clear(): void {
		try {
			dispose(this._toDispose.values());
		} finally {
			this._toDispose.clear();
		}
	}
```